fitbenchmarking/core/fitting_benchmarking.py

```diff
@@ -65,37 +127,204 @@ def fitbenchmark_group(group_name, options, data_dir):
-                  "current options setup and re-run FitBenmarking."
-        raise NoResultsError(message)
 
-    # Used to group elements in list by name
-    results_dict = {}
-    for problem_result in results:
-        name = problem_result.name
-        # group by name
+def loop_over_starting_values(problem, options):
+    """
+    Loops over starting values from the fitting problem
+
+    :param problem: a problem object containing information used in fitting
+    :type problem: FittingProblem
+    :param options: FitBenchmarking options for current run
+    :type options: fitbenchmarking.utils.options.Options
+
+    :return: prob_results array of fitting results for the problem group,
+             list of failed problems and dictionary of unselected minimizers
+    :rtype: tuple(list, list, dict)
+    """
+    name = problem.name
+    num_start_vals = len(problem.starting_values)
+    problem_results = []
+    for index in range(num_start_vals):
+        LOGGER.info("    Starting value: {0}/{1}".format(index + 1,
+                                                         num_start_vals))
+        if num_start_vals > 1:
+            problem.name = name + ', Start {}'.format(index + 1)
+
+        ################################
+        # Loops over fitting softwares #
+        ################################
+        individual_problem_results, problem_fails, unselected_minimzers = \
+            loop_over_fitting_software(problem=problem,
+                                       options=options,
+                                       start_values_index=index)
+        problem_results.extend(individual_problem_results)
+
+    return problem_results, problem_fails, unselected_minimzers
+
+
+def loop_over_fitting_software(problem, options, start_values_index):
+    """
+    Loops over fitting software selected in the options
+
+    :param problem: a problem object containing information used in fitting
+    :type problem: FittingProblem
+    :param options: FitBenchmarking options for current run
+    :type options: fitbenchmarking.utils.options.Options
+
+    :return: list of all results, failed problem names and dictionary of
+             unselected minimizers based on algorithm_type
+    :rtype: tuple(list of fibenchmarking.utils.fitbm_result.FittingResult,
+                  list of failed problem names,
+                  dictionary of minimizers)
+    """
+    grabbed_output = output_grabber.OutputGrabber(options)
+    results = []
+
+    software = options.software
+    if not isinstance(software, list):
+        software = [software]
+
+    problem_fails = []
+    unselected_minimzers = {}
+    software_results = []
+    for s in software:
+        LOGGER.info("        Software: %s", s.upper())
         try:
-            results_dict[name].append(problem_result)
+            minimizers = options.minimizers[s]
         except KeyError:
-            results_dict[name] = [problem_result]
+            raise UnknownMinimizerError(
```

Comment:
Could I suggest `UnsupportedMinimizerError` or something similar?  At first read I thought this meant it was being raised because an unknown error had propagated from the minimizer, rather than because there wasn't a minimizer with that name.

Reply:
Agreed, I have added a new exception and updated the `UnknownMinimizerError` exception as well